database/clickhouse/clickhouse_test.go

```diff
@@ -1,224 +0,0 @@
-package clickhouse_test
-
-import (
-	"context"
-	"database/sql"
-	sqldriver "database/sql/driver"
-	"fmt"
-	"log"
-	"testing"
-
-	_ "github.com/ClickHouse/clickhouse-go"
-	"github.com/dhui/dktest"
-	"github.com/golang-migrate/migrate/v4"
-	"github.com/golang-migrate/migrate/v4/database/clickhouse"
-	dt "github.com/golang-migrate/migrate/v4/database/testing"
-	"github.com/golang-migrate/migrate/v4/dktesting"
-	_ "github.com/golang-migrate/migrate/v4/source/file"
-)
-
-const defaultPort = 9000
-
-var (
-	tableEngines = []string{"TinyLog", "MergeTree"}
-	opts         = dktest.Options{
-		Env:          map[string]string{"CLICKHOUSE_USER": "user", "CLICKHOUSE_PASSWORD": "password", "CLICKHOUSE_DB": "db"},
```

Comment:
This one database `db` both being the default (via env var) *and* the only database ever used by the tested migrations was a little too happy path. It hides the fact that a caller could previously construct migrator instances (e.g., `Open` vs `WithInstance`) in ways that would apply the migrations to the connection's DB while happily creating schema migration metadata tables in the "configured" database.

I've added validations and test cases to make sure this doesn't happen (on either of the v1 or v2 drivers).

The whole thing does make me think it would be cleanest to remove all support for the `clickhouse-go/v1` driver and just use v2 for clarity (it's also what the `clickhouse-go` maintainers recommend anyway).